### ISSUE TITLE
disable:next doesn't work with MainActor attribute in separate line

### ISSUE THREAD

**Issue by chwo** [realm/SwiftLint#5051]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The `disable:next` command doesn't work if `@MainActor` is  in a separate line

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ mint run swiftlint swiftlint

/../FeatureTests.swift:13:7: error: Type Body Length Violation: Type body should span 400 lines or less excluding comments and whitespace: currently spans 538 lines (type_body_length)
/../FeatureTests.swift:12:1: warning: Superfluous Disable Command Violation: SwiftLint rule 'type_body_length' did not trigger a violation in the disabled region; remove the disable command (superfluous_disable_command)
```

### Environment

* SwiftLint version: 0.52.2
* Installation method used: Mint
* Paste your configuration file:

```yml
type_body_length:
  warning: 400 # default '200'
  error: 400 # default '350'
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations): No
* Which Xcode version are you using: 14.3.1 (14E300c)

This doesn't work:

```swift
// swiftlint:disable:next type_body_length
@MainActor
final class FeatureTests: XCTestCase {
    // ...
}
```

This works fine:

```swift
// swiftlint:disable:next type_body_length
@MainActor final class FeatureTests: XCTestCase {
    // ...
}
```

**Comment by mildm8nnered** [comment]
Pretty sure that is "works as expected"

```
@MainActor // swiftlint:disable:next type_body_length
final class FeatureTests: XCTestCase {
    // ...
}
```

is what you'll need if `@MainActor` needs to be on it's own line.

```
@MainActor
final class FeatureTests: XCTestCase { // swiftlint:disable:this type_body_length
    // ...
}
```
and

```
@MainActor
final class FeatureTests: XCTestCase { 
    // swiftlint:disable:previous type_body_length
}
```

Should also work.

**Comment by SimplyDanny** [comment]
@mildm8nnered is right. The disable command must point to the line the issue is triggered on and that's the line where the `class` keyword stands.

Closing. Feel free to object if you don't agree, @chwo.
